src/SFML/Graphics/Image.cpp

```diff
@@ -177,6 +177,11 @@ void Image::copy(const Image& source, const Vector2u& dest, const IntRect& sourc
     if ((source.m_size.x == 0) || (source.m_size.y == 0) || (m_size.x == 0) || (m_size.y == 0))
         return;
 
+    // Make sure the sourceRect left & top  and the {left, top} + {width, height} within bounds
+    if (static_cast<unsigned int>(sourceRect.left) >= source.m_size.x || static_cast<unsigned int>(sourceRect.left + sourceRect.width) > source.m_size.x ||
+        static_cast<unsigned int>(sourceRect.top) >= source.m_size.y || static_cast<unsigned int>(sourceRect.top + sourceRect.height) > source.m_size.y)
+        return;
```

Comment:
I am late, but casting to unsigned int wraps negative values. I don't think that is wanted.
Are negative left and top values for the source rectangle allowed? It is documented not to be allowed since @vittorioromeo's PR #2137, but this is not what this condition checks.

Reply:
I think I was referring to the check at L209

Reply:
If you are referring to this check, `width` and `height` are `unsigned int` at this line, so it is actually checking that both values are not zero.
https://github.com/SFML/SFML/blob/6cd07a043fc36ae75f8f094414d110a5f5333dbf/src/SFML/Graphics/Image.cpp#L208-L210

I was referring to this piece of documentation:

> this function can fail [...] if \a sourceRect is not within the boundaries of the \a source parameter

From this piece of documentation, I understand that `sourceRect` must be inside the rectangle (0, 0, width, height) where width and height are the size of the source image. Do we agree?
If so, I think this implementation must be improved. I will give it a try in the next few days if nobody does it before.

Reply:
That sounds good, thanks.